### ISSUE TITLE
New rule for concat map

### ISSUE THREAD

**Issue by gibiansky** [ndmitchell/hlint#171]
For the following code:

```haskell
ex1 = concat . fmap f
ex2 x = concat $ fmap f x
```

`hlint` should provide the suggestion to use `concatMap`.

**Comment by ndmitchell** [comment]
Seems sensible. Note that adding the hint `concat (fmap f x) == concatMap f x` automatically generates both of those. Want to try a pull request? If not, I'll probably do this before I make the next hlint release.

**Comment by gibiansky** [comment]
Sure, I'll do a PR :) I don't *really* want to but I know that the first PR always decreases activation energy for the next ones, so... :P

**Comment by ndmitchell** [comment]
Should be a trivial one liner in https://github.com/ndmitchell/hlint/blob/master/data/Default.hs, about line 92, so it's a good one to start with :)
